plugins/lua/cxxrandom.lua

```diff
@@ -208,7 +208,7 @@ function num_sequence:shuffle()
     if self.rngID == 'nil' then
         error("Add num_sequence object to crng as distribution, before attempting to shuffle.")
     end
-    ShuffleSequence(self.rngID, self.seqID)
+    ShuffleSequence(self.seqID, self.rngID)
```

Comment:
Why do these classes not use the DFHack class conventions? E.g. `defclass`

Reply:
Dunno what that is.

Reply:
Docs: https://docs.dfhack.org/en/stable/docs/Lua%20API.html#class

Example: https://github.com/DFHack/dfhack/blob/8eb2831b7ed1b3062f9fde1b541da55b89bbf6ac/library/lua/gui.lua#L145

Reply:
It's why I was confused about the `:new()` style the code uses. DFHack classes are normally instantiated like `local obj = My class{namedparam=val, another=val}`

Reply:
I'm like 90% sure I just looked up how to make lua objects and followed whatever SO the search led me to.

Reply:
Well, no need to change it now. Future classes should use the DFHack class structure, tho.